### PR TITLE
gdal 3.7.1

### PR DESCRIPTION
**Pull request by e-n-f**
Replacing stale PR https://github.com/Homebrew/homebrew-core/pull/132095.

Requires:
* https://github.com/Homebrew/homebrew-core/pull/134488 (dronedb)
* https://github.com/Homebrew/homebrew-core/pull/134490 (osmcoastline)
* https://github.com/Homebrew/homebrew-core/pull/134493 (simple-tiles)

- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by chenrui333** [comment]
Just curious why not just merge those three PRs into this one?

**Comment by e-n-f** [comment]
@chenrui333 Because when I did that, I got a "commit style" error that each PR is only allowed to update one package. If you want them all in one, I will be happy to put them all back in one.

**Comment by chenrui333** [comment]
Also to @SMillerDev's point, we need to upstream all these changes so that we can get this PR merged.

**Comment by chenrui333** [comment]
>  I got a "commit style" error that each PR is only allowed to update one package

you just need to make one commit per package, not all changes in one commit (I think that is why you ran into the commit style issue)

**Comment by e-n-f** [comment]
Great, I will merge them all back into this PR.

**Comment by e-n-f** [comment]
Thanks! All four commits should be in this one PR now, and I'll close the others.

**Comment by e-n-f** [comment]
I see that the `dronedb` tests are failing, and will try to figure out what is going wrong.

**Comment by chenrui333** [comment]
Thanks @e-n-f!

File: Formula/simple-tiles.rb [hunk 1]
```diff
@@ -1,10 +1,9 @@
 class SimpleTiles < Formula
   desc "Image generation library for spatial data"
   homepage "https://github.com/propublica/simple-tiles"
-  url "https://github.com/propublica/simple-tiles/archive/v0.6.1.tar.gz"
-  sha256 "2391b2f727855de28adfea9fc95d8c7cbaca63c5b86c7286990d8cbbcd640d6f"
+  url "https://github.com/e-n-f/simple-tiles/archive/refs/tags/v0.7.0.tar.gz"
```
Comment (chenrui333) [inline Formula/simple-tiles.rb:4]: If you are not going to maintain it, we can simply deprecate this formula, no need to cut a new release for gdal

File: Formula/simple-tiles.rb [hunk 2]
```diff
@@ -1,10 +1,9 @@
 class SimpleTiles < Formula
   desc "Image generation library for spatial data"
   homepage "https://github.com/propublica/simple-tiles"
```
Comment (chenrui333) [inline Formula/simple-tiles.rb:3]: this url should match with the artifact.

Comment (e-n-f) [reply Formula/simple-tiles.rb]: OK, fixed

File: Formula/simple-tiles.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/simple-tiles.rb:4]: yeah, make sense.

File: Formula/dronedb.rb [hunk 3]
```diff
@@ -24,6 +24,12 @@ class Dronedb < Formula
   depends_on "libzip"
   depends_on "pdal"
 
+  # To fix gdal-3.7.0
+  patch do
+    url "https://github.com/e-n-f/DroneDB/commit/28aa869dee5920c2d948e1b623f2f9d518bdcb1e.patch?full_index=1"
+    sha256 "50e581aad0fd3226fe5999cc91f9a61fdcbc42c5ba2394d9def89b70183f9c96"
+  end
```
Comment (SMillerDev) [inline Formula/dronedb.rb:31]: Has this been submitted upstream?

Comment (chenrui333) [reply Formula/dronedb.rb:31]: the patch did get merged per this PR, https://github.com/DroneDB/DroneDB/pull/374

**Comment by chenrui333** [comment]
simple-tiles is sorted out, I am gonna wrap up #137944 and bump this to gdal 3.7.1

**Review by chenrui333 — APPROVED** [review]

**Pull request merged**